pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1361,12 +1361,14 @@ func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta
 	// are matched by the incoming pod
 	podToMatchingTermsCount := make(map[string]int)
 	for topologyKey, topologyValue := range node.Labels {
-		if topologyMaps.topologyPairToPods[topologyPair{key: topologyKey, value: topologyValue}] != nil {
-			for existingPod := range topologyMaps.topologyPairToPods[topologyPair{key: topologyKey, value: topologyValue}] {
+		podSet := topologyMaps.topologyPairToPods[topologyPair{key: topologyKey, value: topologyValue}]
+		if podSet != nil {
```

Comment:
Why add this nil check? Ranging over a nil map will simply skip the loop.

Reply:
Good point. Let me update this.